docs/advanced/cmsis-rtos.md

```diff
@@ -0,0 +1,77 @@
+# CMSIS and RTX
+
+CMSIS/RTX code is imported from the original CMSIS repository, which you can find at: https://github.com/ARM-software/CMSIS_5/.
+
+## Memory considerations
+
+Please note that mbed OS doesn't use any of the RTX memory models, which are based on static carveouts (memory pools). This approach is not ideal for generic system, such as mbed O,S because calculating required numbers of RTOS objects is impossible. To avoid declaring arbitrary large buffers, carved out on compile time, limiting amount of available memory, mbed OS shifts the responsibility of supplying the backing memory to CMSIS-RTOS2 users.
```

Comment:
Also probably remove the comma between "large buffers" and "carved out at compile time"